packages/dynamoose/test/types/Item.ts

```diff
@@ -11,3 +11,5 @@ const shouldPassSaveWithReturnItemCallback = user.save({"return": "item"}, () =>
 
 // @ts-expect-error
 const shouldFailWithInvalidReturnType = user.save({"return": "invalid-return-type"});
+
+const shouldPassCustomMethodAccess = user.resetPassword();
```

Comment:
Maybe we should also have a test with a method that doesn't exist? We can use `// @ts-expect-error` for that one.

Reply:
Definitely. Test coverage across edge cases was minimal as I was unsure if you'd be happy with the approach of the PR. Will update/refactor tests to be more comprehensive, and update the documentation as well.